core/src/main/java/org/apache/cxf/bus/spring/BusEntityResolver.java

```diff
@@ -82,7 +82,7 @@ public InputSource resolveEntity(String publicId, String systemId) throws SAXExc
         String resourceLocation = schemaMappings.get(systemId);
         if (resourceLocation != null && publicId == null) {
             Resource resource = new ClassPathResource(resourceLocation, classLoader);
-            if (resource.exists()) {
+            if (resource.exists() && source != null) {
```

Comment:
its better to move null check from L78 to L82

Reply:
@amarkevich  in this case my check for null will not be needed any more . Done with thanks